### ISSUE TITLE
Unable to learn identity mapping

### ISSUE THREAD

**Issue by AvivNavon** [jkalogero/scalegmn#1]
Thank you for this great work and for releasing the code. 

As part of some sanity tests I ran, I tried to train a ScaleGMN model to fit an identity function on MNIST INRs (output the input INR). I tried various configurations, but I couldn't get this to work. I was wondering if you could think of any reason why this could happen or if you are familiar with any such limitation?
Thanks

**Comment by jkalogero** [comment]
Hi @AvivNavon,

Thank you for the interest in our work.

Could you elaborate further about your experiment? Are you comparing the input and output INRs on the function space (in a way similar to the INR editing experiment) or on the weight space?

**Comment by AvivNavon** [comment]
I tried both approaches and also a weighted loss of both function and weight space.

**Comment by jkalogero** [comment]
In case you were using `ScaleGMN_equiv` for your experiment, it seems that there was a bug when converting the graph back to NN parameters, which is now fixed. I have trained ScaleGMN to learn the identity function and it works well for both cases:

- On the _function_ space: Keep in mind to compare the reconstructed output INR with the reconstructed input one (not with the original image as in the INR editing setting).
- On the _parameter_ space: It is not straightforward how to compare the two networks. However, choosing to ignore permutation and scaling symmetries for this simple task seems to work fine. You could use something like:
  ```
  from src.utils.loss import L2_distance
  loss = L2_distance((weights, biases), (new_weights, new_biases)).mean()
  ```
In both cases I only tried a couple of hyperparameter configurations (num_layers:4-5, d_hid:64,128, sign symm/canon), which all seemed to work.

Let me know if you still encounter any issues.

**Comment by AvivNavon** [comment]
Thanks for checking it out @jkalogero! 
I will give it a try again and will update here

**Comment by AvivNavon** [comment]
Amazing, this seems to solve the issue!
Thank you!
